dex/asset.go

```diff
@@ -15,10 +15,52 @@ func (err Error) Error() string { return string(err) }
 
 const (
 	UnsupportedScriptError = Error("unsupported script type")
-	LockTimeTaker          = 24 * time.Hour
-	LockTimeMaker          = 48 * time.Hour
+	lockTimeTaker          = 24 * time.Hour
+	lockTimeMaker          = 48 * time.Hour
 )
 
+var (
+	// These variables are defined to enable setting custom locktime values that
+	// will be used in place of the `lockTimeTaker` and `lockTimeMaker` constants
+	// defined above, _when_ running on a test network (testnet or regtest).
+	// Values for both variables may be set at build time using linker flags e.g:
+	// go build -ldflags "-X 'decred.org/dcrdex/dex.testLockTimeTaker=10m' \
+	// -X 'decred.org/dcrdex/dex.testLockTimeMaker=20m'"
+	// Same values should be set when building server and client binaries.
+	testLockTimeTaker string
+	testLockTimeMaker string
+)
+
+// LockTimeTaker returns the taker locktime value that should be used by both
+// client and server for the specified network. Mainnet uses a constant value
+// while test networks support setting a custom value during build.
+func LockTimeTaker(network Network) time.Duration {
+	if network == Mainnet || testLockTimeTaker == "" {
+		return lockTimeTaker
+	}
+	testLockTime, _ := time.ParseDuration(testLockTimeTaker)
+	if testLockTime.Seconds() == 0 {
+		// Use default value if invalid or zero locktime is specifed.
+		return lockTimeTaker
+	}
+	return testLockTime
+}
+
+// LockTimeMaker returns the taker locktime value that should be used by both
+// client and server for the specified network. Mainnet uses a constant value
+// while test networks support setting a custom value during build.
+func LockTimeMaker(network Network) time.Duration {
```

Comment:
An alternative to these functions would be duration vars that are set by an `init` function in this package.

```go
var(
	testLockTimeTaker string
	testLockTimeMaker string

	lockTimeTaker time.Duration
	lockTimeMaker time.Duration
)

const (
	defaultLockTimeTaker = 24 * time.Hour
	defaultlockTimeMaker = 48 * time.Hour
)

func init() {
	// Parse testLockTimeTaker and testLockTimeMaker if they are not "", setting lockTimeTaker
	// and lockTimeMaker to the result, or panic on parse error
}
```

That way you don't to call these functions repeatedly in each package, and it simplifies NewSwapper.

Reply:
Although there's still the issue of preventing this on Mainnet.  hmm.

Reply:
`init()` won't work because we need to know the current network before setting the `LockTime{Taker,Maker}` values, basically, we'd skip parsing the test values if on mainnet.

Unless... we have 2 set of exported locktime values:
```
const (
	DefaultLockTimeTaker   = 24 * time.Hour
	DefaultLockTimeMaker = 48 * time.Hour
)

var (
	testLockTimeTaker   string
	testLockTimeMaker string

	TestLockTimeTaker   time.Duration
	TestLockTimeMaker time.Duration
)
```

Then `init()` parses the string values and sets `TestLockTimeTaker` and `TestLockTimeMaker` (assuming both `testLockTimeTaker` and `testLockTimeMaker` are not empty strings).

But this would require most usages of `dex.LockTime{T,M}aker` to perform current network checks in determining which locktime values to use:
```
takerLockTime := dex.DefaultLockTimeTaker
if net != dex.Mainnet {
   takerLockTime = dex.TestLockTimeTaker
}
```

Reply:
That's pretty much exactly what I realized in my follow up comment https://github.com/decred/dcrdex/pull/432#discussion_r439472616, except for the exported variable names I was thinking of.  I feel pretty strongly that an unparseable time string should kill the app right away. And I'd prefer not to re-parse the string every time a consumer wants to know the lock time for the network.

For Swapper, if it cannot just access a package-level dex variable, then I'd prefer just to put the two lock time in it's config instead of providing the network just for this purpose since network only was used for the lock times, and the caller can follow the logic of picking lock times based on network, or by calling a function in dex that uses the times already parsed by init.

Reply:
I somehow missed that follow up comment.

I agree that unparseable time string should kill the app right away.

I'd go with the approach of having 2 set of locktime values in the dex package, so the string values are only parsed once in init, and the exported functions just perform the network check.

For Swapper, I'd put the 2 locktimes in it's config so it does not need be aware of the network to do its thing.